backend/utils.py

```diff
@@ -145,5 +168,49 @@ def format_stream_response(chatCompletionChunk, history_metadata, message_uuid=N
                     }
                     response_obj["choices"][0]["messages"].append(messageObj)
                     return response_obj
-    
-    return {}
\ No newline at end of file
+
+    return {}
+
+
+def convert_to_pf_format(input_json):
+    output_json = []
+    # align the input json to the format expected by promptflow chat flow
+    for message in input_json["messages"]:
+        if message:
+            if message["role"] == "user":
+                new_obj = {
+                    "inputs": {"question": message["content"]},
+                    "outputs": {"answer": ""},
+                }
+                output_json.append(new_obj)
+            elif message["role"] == "assistant" and output_json:
+                output_json[-1]["outputs"]["answer"] = message["content"]
+    return output_json
+
+
+def get_pf_swagger_spec():
+    logging.debug("Fetching Promptflow Swagger spec")
+    response = requests.get(
+        url=f"{os.environ.get('PROMPTFLOW_ENDPOINT').split('/score')[0]}/swagger.json",
+        headers={
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+            "Authorization": "Bearer " + os.environ.get("PROMPTFLOW_API_KEY"),
+        },
+    )
+    if response.status_code == 200:
+        swagger_dict = response.json()  # Convert the response to JSON
+        response_fields = swagger_dict["paths"]["/score"]["post"]["responses"]["200"][
```

Comment:
Do we know all of the possible ways this could change / break in the future?  I'm just wondering if we could be defensive here to handle any `KeyErrors` that might occur if there are any structural changes to the spec on the Promptflow side.